src/main/java/com/example/springbootrestexample/logic/BookLogic.java

```diff
@@ -0,0 +1,81 @@
+package com.example.springbootrestexample.logic;
+
+import com.example.springbootrestexample.comparators.BookIsbnComparator;
+import com.example.springbootrestexample.comparators.BookNameComparator;
+import com.example.springbootrestexample.exceptions.BookIsbnDoesNotExistException;
+import com.example.springbootrestexample.models.Book;
+import com.example.springbootrestexample.repositories.AuthorRepository;
+import com.example.springbootrestexample.repositories.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class BookLogic extends ExceptionMessageLogic {
+
+    @Autowired
+    private BookRepository bookRepository;
+    @Autowired
+    private AuthorRepository authorRepository;
+
+    public List<Book> getAllBooks(){
+        return getAllBooks("");
+    }
+
+    public List<Book> getAllBooks(String columnOrder){
+        List<Book> books = bookRepository.listBooks();
+
+        switch (columnOrder){
```

Comment:
Can also use a Enum, enum.valueOf(columOrder) and you do not have those strings hardcoded here